### PR TITLE
Add safeguards to bbolt CLI tool

### PR DESCRIPTION
**Pull request by ptabor**
- perform sanity check whether pages being fetched are bbolt pages and not random data (e.g. they self identify using proper magic number of 'pageid' field. 
- protect against alocation of extreeme emount of RAM if 'overflow pages' field contains too much data. 

Signed-off-by: Piotr Tabor <ptab@google.com>

### CODE REVIEW

File: cmd/bbolt/main.go [hunk 1]
```diff
@@ -1796,46 +1799,62 @@ func ReadPage(path string, pageID int) (*page, []byte, error) {
 
 	// Determine total number of blocks.
 	p := (*page)(unsafe.Pointer(&buf[0]))
+	if p.id != pgid(pageID) {
+		return nil, nil, fmt.Errorf("error: %w due to unexpected page id: %d != %d", ErrCorrupt, p.id, pageID)
+	}
 	overflowN := p.overflow
+	if overflowN >= uint32(hwm) {
```
Comment (ahrtr) [inline cmd/bbolt/main.go:1806]: We need to exclude the 2 meta pages and current page. For example, assuming there are 6 pages in total, then the overflowN field of any page should be <= 3

```suggestion
	if overflowN >= uint32(hwm) - 3 {
```

Comment (ptabor) [reply cmd/bbolt/main.go:1806]: Corrected. I didn't intended to be that precise - just protect OS from extensive allocation.

**Review by ahrtr — APPROVED** [review]
LGTM

**Comment by ahrtr** [comment]
could you please add a changelog item in [CHANGELOG.md](https://github.com/etcd-io/bbolt/blob/master/CHANGELOG.md)?

**Pull request merged**